### PR TITLE
sys/posix: rename posix_time to posix_sleep

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR is a refactoring of the `posix_time` module. In master, this module provides an implementation of POSIX `sleep` and `usleep` functions (see [here](https://pubs.opengroup.org/onlinepubs/9699919799/functions/sleep.html)). This is confusing since POSIX also defines functions for manipulating time (see [time.h](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/time.h.html)).
The [nanosleep](https://pubs.opengroup.org/onlinepubs/9699919799/functions/nanosleep.html) is also a candidate for this module since it should be used as a replacement of `usleep` (which was removed in POSIX.1-2008). This is not part of this PR and is left as a follow-up.

Other changes come with this PR:
- the module is split in 2, now there are `posix_sleep` (for the sleep function only) and `posix_usleep`
- the corresponding tests are split as well
- ztimer is used as a replacement of xtimer (initial motivation of this PR). The split in 2 separate modules allows to use the rtt backend (and thus low-power) with the sleep function.

When adding ztimer, I noticed the `posix_sleep` test was failing because `ztimer_sleep(ZTIMER_MSEC, 0);` sleeps during ~2s (on STM32L4). I added 1f5cf14 to work-around that but I'm wondering if a simple return shouldn't be added in `ztimer_sleep` if duration is zero.

### Testing procedure

- A green Murdock (with `CI: run tests`)

### Issues/PRs references

None

### CODE REVIEW

File: makefiles/pseudomodules.inc.mk [hunk 1]
```diff
@@ -95,6 +95,8 @@ PSEUDOMODULES += picolibc
 PSEUDOMODULES += picolibc_stdout_buffered
 PSEUDOMODULES += pktqueue
 PSEUDOMODULES += posix_headers
+PSEUDOMODULES += posix_sleep
+PSEUDOMODULES += posix_usleep
```
Comment (miri64) [inline makefiles/pseudomodules.inc.mk:99]: Why two modules? Does having both functions in one module really generate such a huge overhead?

Comment (aabadie) [reply makefiles/pseudomodules.inc.mk:99]: See third point of PR changes in OP: `The split in 2 separate modules allows to use the rtt backend (and thus low-power) with the sleep function.`

Comment (miri64) [reply makefiles/pseudomodules.inc.mk:99]: So you can now only use either `posix_sleep` or `posix_usleep`?

Comment (aabadie) [reply makefiles/pseudomodules.inc.mk:99]: > So you can now only use either posix_sleep or posix_usleep?

Yes. What do you mean with this question ? You want 2 separate submodules and no pseudo-modules ?

Comment (miri64) [reply makefiles/pseudomodules.inc.mk:99]: I was wondering, 

1. If having two modules is sensible with regards of POSIX usability (we don't have a subdivision in multiple modules for other POSIX functions either) and
2. With this new information you gave me, if this is POSIX compliant and practical. What if a user needs both functions when porting a library? Can't there be at least an adaption layer to use `posix_usleep()` for `posix_sleep()` when both `posix_sleep` and `posix_usleep` are compiled in so you still can use both?

Comment (aabadie) [reply makefiles/pseudomodules.inc.mk:99]: > What if a user needs both functions when porting a library?

```
USEMODULE += posix_sleep
USEMODULE += posix_usleep
```
also works. I need to check the effects on the underlying timers (`ZTIMER_USEC` and `ZTIMER_MSEC`). While usec will use a high-speed timer backend without low-power capabilities, I'm unsure if msec will still use the rtt backend. Maybe yes, and in this case there would be no need for the split `posix_sleep`/`posix_sleep`, which would simplify this PR. @kaspar030 what do you think ?

Comment (miri64) [reply makefiles/pseudomodules.inc.mk:99]: Now I'm confused :sweat_smile:. Are those modules mutually exclusive or not?

Comment (aabadie) [reply makefiles/pseudomodules.inc.mk:99]: > Are those modules mutually exclusive or not?

After discussing with @fjmolinas IRL, this is not the case. So my initial assumption, which lead to the split in 2 modules was wrong. I'll rework that PR.

**Comment by aabadie** [comment]
Reworked that PR to put sleep and usleep in the same posix_sleep module. The test application is not split anymore. I also noticed that 1f5cf14df010e9069ccf92884aa1a01d90c22b1a was not needed, the test is working like a charm on b-l475e-iot01a. The issue I had is probably related to the RTT config of the nucleo-l412kb (that I used for testing).

**Comment by aabadie** [comment]
Argh, `posix_sleep` doesn't fit anymore on nucleo-l011k4: `region `ram' overflowed by 8 bytes`

**Comment by aabadie** [comment]
Are you ok with the updates of this PR @miri64 ?

**Comment by miri64** [comment]
> Argh, `posix_sleep` doesn't fit anymore on nucleo-l011k4: `region 'ram' overflowed by 8 bytes`

Might be reversible once everything runs on `ztimer` :-).

**Review by miri64 — APPROVED** [review]
> Are you ok with the updates of this PR @miri64 ?

Yepp. Just changes the name now and ports the module to `ztimer`.

**Pull request merged**